### PR TITLE
Add React.SFC type for Form.create return value #8672

### PR DESCRIPTION
**Pull request by kunlongxu**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: components/form/Form.tsx [hunk 1]
```diff
@@ -10,188 +10,188 @@ import FormItem from './FormItem';
...
 
 export interface FormCreateOption<T> {
-  onFieldsChange?: (props: T, fields: Array<any>) => void;
-  onValuesChange?: (props: T, values: any) => void;
-  mapPropsToFields?: (props: T) => void;
-  withRef?: boolean;
+    onFieldsChange?: (props: T, fields: Array<any>) => void;
+    onValuesChange?: (props: T, values: any) => void;
+    mapPropsToFields?: (props: T) => void;
+    withRef?: boolean;
```
Comment (yesmeck) [inline components/form/Form.tsx]: Don't change indent.

Comment (linmodev) [reply components/form/Form.tsx]: Oh,My editor changed the indent.

File: components/form/Form.tsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ export type GetFieldDecoratorOptions = {
   exclusive?: boolean;
   /** Normalize value to form component */
   normalize?: (value: any, prevValue: any, allValues: any) => any;
-  /** Whether stop validate on first rule of error for this field.	 */
+  /** Whether stop validate on first rule of error for this field.   */
```
Comment (afc163) [inline components/form/Form.tsx]: No need to change this.

**Comment by afc163** [comment]
CI broken.

File: components/form/Form.tsx [hunk 3]
```diff
@@ -70,7 +70,7 @@ export type GetFieldDecoratorOptions = {
   exclusive?: boolean;
   /** Normalize value to form component */
   normalize?: (value: any, prevValue: any, allValues: any) => any;
-  /** Whether stop validate on first rule of error for this field.	 */
+  /** Whether stop validate on first rule of error for this field.  */
   validateFirst?: boolean;
```
Comment (linmodev) [inline components/form/Form.tsx]: What is the difference between these two lines?

Comment (valleykid) [reply components/form/Form.tsx]: blank spaces?

Comment (linmodev) [reply components/form/Form.tsx]: No, I changed the two lines have the same number of spaces, but still different, now has passed, it does not matter

Comment (linmodev) [reply components/form/Form.tsx]: No, I changed the two lines have the same number of spaces, but still different, now has passed, it does not matter

**Pull request merged**